src/pykeen/training/training_loop.py

```diff
@@ -401,6 +411,15 @@ def _train(  # noqa: C901
         elif not self.model.supports_subbatching:
             raise SubBatchingNotSupportedError(self.model)
 
+        model_contains_batch_norm = any(
+            isinstance(m, (nn.BatchNorm1d, nn.BatchNorm2d, nn.BatchNorm3d, nn.SyncBatchNorm))
+            for m in self.model.modules()
+        )
+        if batch_size == 1 and model_contains_batch_norm:
+            raise ValueError("Cannot train a model with batch_size=1 containing BatchNorm layers.")
+        if drop_last is None:
+            drop_last = model_contains_batch_norm
```

Comment:
Shall we add an info message or warning?

Reply:
650cfd6